services/dataplatform/cluster.go

```diff
@@ -129,9 +132,48 @@ func setPatchClusterRequestProperties(d *schema.ResourceData) *dataplatform.Patc
 		dataplatformCluster.Properties.MaintenanceWindow = setMaintenanceWindowData(d)
 	}
 
+	dataplatformCluster.Properties.Lans = setLansData(d)
+
 	return &dataplatformCluster
 }
 
+func setLansData(d *schema.ResourceData) *[]dataplatform.Lan {
```

Comment:
do we need the pointer to the slice here? Can't we just return just the slice?

Reply:
yes, we do:
```
dataplatformCluster.Properties.Lans = setLansData(d)
```
where `Lans` is a pointer to the slice:
```
Lans *[]Lan `json:"lans,omitempty"`
```
Same goes for all the similar comments, for `routes` for example.